frontend/src/components/form/FormAutosave.vue

```diff
@@ -0,0 +1,42 @@
+<script setup lang="ts">
+import { useFormValues, useIsFormDirty } from 'vee-validate';
+import { ref, watch } from 'vue';
+
+import type { Ref } from 'vue';
+
+// Props
+const { callback, delay = 10000 } = defineProps<{
```

Comment:
Optional style: May be worth having the timeout delay be a named const value instead of as a magic number for improving readability.